packages/video_player/video_player/lib/video_player.dart

```diff
@@ -634,6 +633,16 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
     return Caption.none;
   }
 
+  /// Sets a closed caption file.
+  ///
+  /// If [closedCaptionFile] is null, closed caption file will be removed.
+  Future<void> setClosedCaptionFile(
+    Future<ClosedCaptionFile>? closedCaptionFile,
```

Comment:
I think this would be better as `FutureOr<ClosedCaptionFile?> closedCaptionFile ` so a user wouldn't have to use `Future.value()` to pass a value synchronously.

Reply:
Or better yet, I think just making it synchronous would be best if possible: `ClosedCaptionFile? closedCaptionFile`.

Reply:
Actually, I made it synchronous but then I turned it to async because closedCaptionFile parameter in initialize method is async. I don't know why closedCaptionFile parameter is Future but I thought if they made it Future, there was a logic behind it. However, my main thought is to make it synchronous as I mentioned in the issue (In short, removing closedCaptionFile parameter and adding a sync setter and getter for closedCaptionFile). But this would be a breaking change, so I decided to add just a setter method. I discussed this with stuartmorgan but I want to ask that question to you too because I'm wondering your thoughts, Can we make this breaking change? I think synchronous would be better in general.

Reply:
I missed in my review that you didn't change `final Future<ClosedCaptionFile>? closedCaptionFile;` to a getter, which was part of what we discussed. This change will leave the public accessor returning the wrong file.

(That should also be added to the tests, since it was missed.)

> Can we make this breaking change

We are not making a breaking change in `video_player` for this; that's a disruptive change, and this is a niche use case that can easily be accommodated by a non-breaking change.

@bparrishMines I forgot the previous discussion was in Discord rather than here, so you wouldn't have context, sorry about that. I suggested the `Future<>` version for consistency; I could see an argument for making it take a file directly and then convert to a Future internally for compatibility with the existing public getter API; I don't have a strong opinion on that.

Reply:
Ah, I hadn't realized this was already discussed. No worries. This solution sounds good to me! (Assuming @stuartmorgan will wait for the accessor change.)

Reply:
> I missed in my review that you didn't change `final Future<ClosedCaptionFile>? closedCaptionFile;` to a getter, which was part of what we discussed. This change will leave the public accessor returning the wrong file.

Should I do that non-final? If I do that non-final they are be able to set without `setClosedCaptionFile`. Like this: `controller.closedCaptionFile = null;`. But if it is not a big deal I can do that.

Reply:
A getter does not allow assigning a value (nor is the `final` keyword applicable to it) so I don't understand the question.

Reply:
> A getter does not allow assigning a value (nor is the `final` keyword applicable to it)
Yes, I know. 

To be honest, I don't know how to change `final Future<ClosedCaptionFile>? closedCaptionFile;` to a getter. I changed it to `Future<ClosedCaptionFile>? closedCaptionFile;` but it became assignable. That's what i'm trying to say. So, how can I change it to a getter?

Reply:
I believe @stuartmorgan was thinking something like:

```dart
VideoPlayerController.asset(/*other params*/, {Future<closedCaptionFile>? closedCaptionFile, /*otherparams*/})
      : _closedCaptionFile = closedCaptionFile;

Future<ClosedCaptionFile>? _closedCaptionFile;

Future<ClosedCaptionFile>? get closedCaptionFile {
  return _closedCaptureFile;
}
```

Reply:
Exactly, with the caveat that the member variable will need to be called something like `_closedCaptionFileFuture` since there's already a `_closedCaptionFile`.

> To be honest, I don't know how to change `final Future<ClosedCaptionFile>? closedCaptionFile;` to a getter. I turned it to `Future<ClosedCaptionFile>? closedCaptionFile;` but it became assignable. That's what i'm trying to say.

In the future, if you aren't sure how to implement a suggestion it is much more effective to simply ask, rather than doing something completely different without explaining what you did and then ask questions about that instead. I had absolutely no context for the question you were asking since I didn't know what the code you'd actually written was (or that it was different from my suggestion).

Reply:
> Exactly, with the caveat that the member variable will need to be called something like _closedCaptionFileFuture since there's already a _closedCaptionFile.

Thanks for the answer. I implemented this and everything works fine except the document. Now, the closedCaptionFile parameter shows the constructors' document. (I'm not talking about the getter, its document is ok)

> In the future, if you aren't sure how to implement a suggestion it is much more effective to simply ask, rather than doing something completely different without explaining what you did and then ask questions about that instead. I had absolutely no context for the question you were asking since I didn't know what the code you'd actually written was (or that it was different from my suggestion).

Thanks for your feedback. I'll do.

Reply:
I committed the changes